### PR TITLE
Remove CONDITION from event object completely

### LINKED ISSUE #3172: Remove CONDITION from event completely
The output of the "kubectl get events" command is currently confusing and, at times, misleading. After talking to a few folks, most of the confusion seems to arise from the ```CONDITION``` column:
* It is not clear how the possible values (```waiting```, ```running```, ```Pending```, ```failed```, etc.) for the ```CONDITION``` column are determined
  * Often the code where an event is generated does not have enough information to accurately make this determination and thus assigns the value arbitrarily.
  * Because values are assigned arbitrarily, the meaning is often unclear.
* The information provided in the ```REASON``` column seems sufficient to understanding what "phase/state transition" the event is reporting.

Proposal:
* Remove the ```CONDITION``` column from "kubectl get events" output

Current sample output, for reference:
```
cluster/kubectl.sh get events

TIME                              NAME                                          KIND                CONDITION           REASON              MESSAGE
Mon, 29 Dec 2014 21:56:05 +0000   kubernetes-minion-1.c.saad-dev-vms.internal   Minion                                  starting            Starting kubelet.
Mon, 29 Dec 2014 21:56:59 +0000   influx-grafana                                BoundPod            waiting             pulled              Successfully pulled image kubernetes/pause:latest
Mon, 29 Dec 2014 21:56:59 +0000   influx-grafana                                BoundPod            waiting             created             Created with docker id 0536c9d30742adf4a6a956660ec99b2b71dcc68a478a35aa193fdea807951762
Mon, 29 Dec 2014 21:56:59 +0000   influx-grafana                                Pod                 Pending             scheduled           Successfully assigned influx-grafana to kubernetes-minion-1.c.saad-dev-vms.internal
Mon, 29 Dec 2014 21:57:00 +0000   influx-grafana                                BoundPod            running             started             Started with docker id 0536c9d30742adf4a6a956660ec99b2b71dcc68a478a35aa193fdea807951762
Mon, 29 Dec 2014 21:57:03 +0000   heapster                                      Pod                 Pending             scheduled           Successfully assigned heapster to kubernetes-minion-1.c.saad-dev-vms.internal
Mon, 29 Dec 2014 21:57:03 +0000   heapster                                      BoundPod            waiting             pulled              Successfully pulled image kubernetes/pause:latest
Mon, 29 Dec 2014 21:57:04 +0000   heapster                                      BoundPod            waiting             created             Created with docker id a5e026c56ab62408b1246ec1e5a8d04156dd409422c35c7a783bc60a18546d2a
Mon, 29 Dec 2014 21:57:04 +0000   heapster                                      BoundPod            running             started             Started with docker id a5e026c56ab62408b1246ec1e5a8d04156dd409422c35c7a783bc60a18546d2a
Mon, 29 Dec 2014 21:57:08 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             created             Created with docker id 354b0bbc5e74ff03ca77c202a4ac6994eb54444701d2a77c7ec6f896f1e76d78
Mon, 29 Dec 2014 21:57:08 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             pulled              Successfully pulled image kubernetes/pause:latest
Mon, 29 Dec 2014 21:57:09 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          Pod                 Pending             scheduled           Successfully assigned a20d5f23-8fa5-11e4-8514-42010af08855 to kubernetes-minion-1.c.saad-dev-vms.internal
Mon, 29 Dec 2014 21:57:09 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            running             started             Started with docker id 354b0bbc5e74ff03ca77c202a4ac6994eb54444701d2a77c7ec6f896f1e76d78
Mon, 29 Dec 2014 21:57:22 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             pulled              Successfully pulled image quay.io/coreos/etcd:latest
Mon, 29 Dec 2014 21:57:23 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            running             started             Started with docker id 3dca34da5bd7969ce6678daad022a40f1887aac26a1f075ff5a83caab31d8f51
Mon, 29 Dec 2014 21:57:23 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             created             Created with docker id 3dca34da5bd7969ce6678daad022a40f1887aac26a1f075ff5a83caab31d8f51
Mon, 29 Dec 2014 21:58:06 +0000   influx-grafana                                BoundPod            failed              failed              Failed to pull image kubernetes/heapster_influxdb
Mon, 29 Dec 2014 21:58:07 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             pulled              Successfully pulled image kubernetes/kube2sky:1.0
Mon, 29 Dec 2014 21:58:07 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             created             Created with docker id 431ecef5c439c0e79f23e3cc2f365623d02cf3e19267866a2c135fe00711f33a
Mon, 29 Dec 2014 21:58:08 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            running             started             Started with docker id 431ecef5c439c0e79f23e3cc2f365623d02cf3e19267866a2c135fe00711f33a
Mon, 29 Dec 2014 21:58:14 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            running             started             Started with docker id 4a4ba2d9edee94dcfe37ff878346f348aefc4980d3ca0bcb46aa33ee727c1750
Mon, 29 Dec 2014 21:58:14 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             created             Created with docker id 4a4ba2d9edee94dcfe37ff878346f348aefc4980d3ca0bcb46aa33ee727c1750
Mon, 29 Dec 2014 21:58:14 +0000   a20d5f23-8fa5-11e4-8514-42010af08855          BoundPod            waiting             pulled              Successfully pulled image kubernetes/skydns:2014-12-23-001
Mon, 29 Dec 2014 21:58:21 +0000   heapster                                      BoundPod            waiting             pulled              Successfully pulled image kubernetes/heapster
Mon, 29 Dec 2014 21:58:21 +0000   heapster                                      BoundPod            running             started             Started with docker id af873a78cad7e8ddb4550db73325893988c0e6f3f6708e3d50dc8d7d97771609
Mon, 29 Dec 2014 21:58:21 +0000   heapster                                      BoundPod            waiting             created             Created with docker id af873a78cad7e8ddb4550db73325893988c0e6f3f6708e3d50dc8d7d97771609
Mon, 29 Dec 2014 21:59:10 +0000   influx-grafana                                BoundPod            failed              failed              Failed to pull image kubernetes/heapster_grafana
Mon, 29 Dec 2014 21:59:59 +0000   influx-grafana                                BoundPod            waiting             created             Created with docker id 7a1f296a6b1602f18bd5911d73b40a2b356b7a871167539f2da941254a9cc7ae
Mon, 29 Dec 2014 21:59:59 +0000   influx-grafana                                BoundPod            waiting             pulled              Successfully pulled image dockerfile/elasticsearch
Mon, 29 Dec 2014 21:59:59 +0000   influx-grafana                                BoundPod            running             started             Started with docker id 7a1f296a6b1602f18bd5911d73b40a2b356b7a871167539f2da941254a9cc7ae
Mon, 29 Dec 2014 22:00:50 +0000   influx-grafana                                BoundPod            running             started             Started with docker id 49a3b1b6918af2233db7049d70362af4db9a1a3ae2dea930d79ad921781dc690
Mon, 29 Dec 2014 22:00:50 +0000   influx-grafana                                BoundPod            waiting             created             Created with docker id 49a3b1b6918af2233db7049d70362af4db9a1a3ae2dea930d79ad921781dc690
Mon, 29 Dec 2014 22:00:50 +0000   influx-grafana                                BoundPod            waiting             pulled              Successfully pulled image kubernetes/heapster_influxdb
```

### PR DESCRIPTION
**Pull request by saad-ali**
Closes #3172 

Removes the `CONDITION` column from "kubectl get events" output and from the `Event` object
See #3172 for rational

### CODE REVIEW

**Comment by thockin** [comment]
I am good with this.  @lavalamp ?

File: pkg/api/v1beta1/types.go [hunk 1]
```diff
@@ -825,7 +826,7 @@ type Event struct {
 	// TODO: provide exact specification for format.
 	Source string `json:"source,omitempty" description:"component reporting this event; short machine understandable string"`
 	// Host name on which the event is generated.
-	Host string `json:"host,omitempty"`
+	Host string `json:"host,omitempty" description:"name on which the event is generated"`
```
Comment (lavalamp) [inline pkg/api/v1beta1/types.go]: location of component reporting this event?

File: pkg/client/record/event.go [hunk 2]
```diff
@@ -127,17 +127,16 @@ var events = watch.NewBroadcaster(queueLen)
 // Event constructs an event from the given information and puts it in the queue for sending.
 // 'object' is the object this event is about. Event will make a reference-- or you may also
 // pass a reference to the object directly.
-// 'condition' is the new condition of the object. 'reason' is the reason it now has this status.
-// Both 'condition' and 'reason' should be short and unique; they will be used to automate
-// handling of events, so imagine people writing switch statements to handle them. You want to
-// make that easy.
+// 'reason' is the reason it now has this status. 'reason' should be short and unique; it will
```
Comment (lavalamp) [inline pkg/client/record/event.go]: reword without reference to status?

**Comment by lavalamp** [comment]
LGTM-- just a few nits

**Comment by saad-ali** [comment]
Nits addressed. Commits squashed.

**Comment by saad-ali** [comment]
E2E looks good:
2015/01/14 14:05:33 Passed tests: basic.sh[1/1] certs.sh[1/1] goe2e.sh[1/1] guestbook.sh[1/1] liveness.sh[1/1] monitoring.sh[1/1] pd.sh[1/1] private.sh[1/1] services.sh[1/1] update.sh[1/1]
2015/01/14 14:05:33 Flaky tests:
2015/01/14 14:05:33 Failed tests:
2015/01/14 14:05:33 Success!

**Comment by lavalamp** [comment]
LGTM, will merge when travis finishes.

**Pull request merged**